src/libs/vtkh/filters/Filter.cpp

```diff
@@ -108,16 +108,50 @@ Filter::PostExecute()
 void
 Filter::MapAllFields()
 {
-  if(m_input->GetNumberOfDomains() > 0)
-  {
-    vtkm::cont::DataSet dom = m_input->GetDomain(0);
-    vtkm::IdComponent num_fields = dom.GetNumberOfFields();
-    for(vtkm::IdComponent i = 0; i < num_fields; ++i)
+    int num_domains = m_input->GetNumberOfDomains();
+    if(num_domains > 0)
     {
-      std::string field_name = dom.GetField(i).GetName();
-      m_map_fields.push_back(field_name);
+        if(num_domains > 1)
+        {
+
+            vtkm::cont::DataSet dom_0 = m_input->GetDomain(0);
```

Comment:
Need to think about this one, is this logic to figure out of fields related to a toplogy should be mapped?

Reply:
vtkm lets you grab fields by name or index 

I can see about adding a union of fields names

Reply:
Thanks -- the index (and insertion order) could be important then, so we could try the union approach, but with a vector instead of a set.  Insert into vector at back if entry is not already in there.  Still not perfect - the best thing would be a per-domain list that is applied downstream, but I don't think we are setup for that.

We call GetFieldSelection and pass that entire list to VTK-m (there is no per domain field selection plumbing)